### ISSUE TITLE
Moon phase information isn't read from simspec files in  /scripts/quickgen.py

### ISSUE THREAD

**Issue by akremin** [desihub/desisim#260]
@sbailey @moustakas 

The code checks for explicit inputs for moon parameters, but doesn't consult the simspec file. It's a simple change that I would have made myself, but I think a more senior member should make a choice of where these things should be defined. I was under the impression they should be defined in simspec files created by new_exposure (or some variant). Maybe the ability to define them in quickgen has a use case, but it does seemto leave room for overwriting what people think they are defining when making the simspec files.

Lines 441-480.  
 `    
#### Moon Phase Information
    if flavor in bright_objects or object_type in bright_objects:
        if args.moon_phase is None:
            qsim.atmosphere.moon.moon_phase = 0.7
        else:
            qsim.atmosphere.moon.moon_phase = args.moon_phase
        if args.moon_angle is None:
            qsim.atmosphere.moon.separation_angle = 50 * u.deg
        else:
            qsim.atmosphere.moon.separation_angle = args.moon_angle * u.deg
        if args.moon_zenith is None:
            qsim.atmosphere.moon.moon_zenith = 30 * u.deg
        else:
            qsim.atmosphere.moon.moon_zenith = args.moon_zenith * u.deg
    elif flavor in gray_objects:
        if args.moon_phase is None:
            qsim.atmosphere.moon.moon_phase = 0.1
        else:
            qsim.atmosphere.moon.moon_phase = args.moon_phase
        if args.moon_angle is None:
            qsim.atmosphere.moon.separation_angle = 60 * u.deg
        else:
            qsim.atmosphere.moon.separation_angle = args.moon_angle * u.deg
        if args.moon_zenith is None:
            qsim.atmosphere.moon.moon_zenith = 80 * u.deg
        else:
            qsim.atmosphere.moon.moon_zenith = args.moon_zenith * u.deg
    else:
        if args.moon_phase is None:
            qsim.atmosphere.moon.moon_phase = 0.5
        else:
            qsim.atmosphere.moon.moon_phase = args.moon_phase
        if args.moon_angle is None:
            qsim.atmosphere.moon.separation_angle = 60 * u.deg
        else:
            qsim.atmosphere.moon.separation_angle = args.moon_angle * u.deg
        if args.moon_zenith is None:
            qsim.atmosphere.moon.moon_zenith = 100 * u.deg
        else:
qsim.atmosphere.moon.moon_zenith = args.moon_zenith * u.deg`

**Comment by sbailey** [comment]
Good catch.  The intension was:
```
if user specifies lunar parameters on the command line:
    use those command line params, overriding what might be in the simspec file
elif lunar parameters exist in the simspec files:
    use those simspec parameters
else:
    fall back to defaults per (dark, gray, bright) flavors only as a last resort
```

Does that make sense and can you update quickgen to support that?

**Comment by akremin** [comment]
Yes that's clear. It looks like something very similar is already there for
the exposure time.

Yes I can make the change.

Best,
Anthony

**Comment by sbailey** [comment]
I now think that the implementation for airmass and exposure time is buggy.  Command line options should trump values in the files if the option is specified.  And we should be consistent for what method trumps what across all parameters.

In the future, the "normal" path should be to have the parameters in the simspec file and use that without having to override them from the command line.  The command line just provides a way to override it for special studies without having to create N>>1 otherwise identical simspec files.

**Comment by akremin** [comment]
I agree 100%.

Just wanted to make sure we are on the same page.

I'll make the changes, which now include airmass and exposure times, in
addition to the moon parameters.

Best,
Anthony

**Comment by moustakas** [comment]
Fixed in #263.
